packages/mobile/src/fiatExchanges/MoonPay.tsx

```diff
@@ -0,0 +1,64 @@
+import colors from '@celo/react-components/styles/colors'
+import * as React from 'react'
+import { ActivityIndicator, StyleSheet, View } from 'react-native'
+import { WebView } from 'react-native-webview'
+import { useSelector } from 'react-redux'
+import { showError } from 'src/alert/actions'
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import { SIGN_MOONPAY_FIREBASE_URL } from 'src/config'
+import { getLocalCurrencyCode } from 'src/localCurrency/selectors'
+import { currentAccountSelector } from 'src/web3/selectors'
+
+const celoCurrencyCode = 'CUSD'
+
+function FiatExchangeWeb() {
+  const [uri, setUri] = React.useState('')
+  const account = useSelector(currentAccountSelector)
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+  React.useEffect(() => {
```

Comment:
what do you think about defining this custom hook outside of the function? 

or having a getSignedURL function outside of the hook that takes in `amount, celoCurrencyCode, account, localCurrencyCode as arguments` ? 

or both?

Reply:
What would be the benefits of doing it this way?

Reply:
the intention would be a smaller function size. 

I just think its a nice way to organize :)